SeaBattle/Ship.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SeaBattle
+{
+    public class BaseShip
+    {
+        public BaseShip (string Name, List<PointStruct> points) { }
+        private List<Deck> Decks { get; set; }
+        int Durability;
+        int MoveSpeed;
+        int MoveDirectionX;
+        int MoveDirectionY;
+
+    }
+
+    public class Deck
+    {
+        public PointStruct Location;
+
+        // If 0 = SANK, 1 = AFLOAT.
+        public int State;
+    }
+
+    interface IAttack
+    {
+        void Shoot();
+    }
+    interface IRepair
+    {
+        void Repair();
+    }
+
+    public class BattleShip : BaseShip, IAttack
+    {
+        public BattleShip(string Name, List<PointStruct> points) : base(Name, points) 
+        {
+
+            var Decks = new List<Deck>();
+
+            for (int i = 0; i < points.Count - 1; i++)
+            {
+                Deck a = new Deck();
+                a.Location = points[i];
+                Decks.Add(a);
+                a.State = 1;
+            }
+
+        }
+
+        void IAttack.Shoot() { 
```

Comment:
How is this even gonna work?

Reply:
It is not finished parts of interfaces.
This will work BattleShip.Shoot(); ServiceShip.Repair(); etc.

Reply:
you are using here Explicit Interface Implementation, why?
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation